scio-core/src/main/scala/com/spotify/scio/io/Tap.scala

```diff
@@ -71,9 +72,10 @@ case class TextTap(path: String) extends Tap[String] {
  * @param schema must be not null if `T` is of type
  *               [[org.apache.avro.generic.GenericRecord GenericRecord]].
  */
-case class AvroTap[T: ClassTag](path: String, schema: Schema = null) extends Tap[T] {
-  override def value: Iterator[T] = FileStorage(path).avroFile(schema)
-  override def open(sc: ScioContext): SCollection[T] = sc.avroFile[T](path, schema)
+case class AvroTap[T: ClassTag](path: String, @transient schema: Schema = null) extends Tap[T] {
```

Comment:
`schema` will be null after the tap goes through ser/de right? That breaks the contract. And the user might access it accidentally.

Can't think of a way to fit this in a case class. Make it a boring class, `_schema: Schema` a ctor arg, and `def schema: Schema` to safely get it? Or is there any better way?